File: packages/python/qsu/file/isFileExists.py
import os


def isFileExists(filePath: str) -> bool:
	try:
		# `exists` follows symlinks, so a dangling link reports as missing. The
		# `access` call that used to sit in front of this one had its result
		# thrown away, and cost a system call for nothing.
		return os.path.exists(filePath)
	except Exception:
		return False

File: packages/javascript/lib/node/file/isFileExists.ts
import { stat } from 'fs/promises';

export async function isFileExists(filePath: string): Promise<boolean> {
	try {
		// `stat` follows symlinks, so a dangling link reports as missing. This
		// matches Dart's `File.exists` and Python's `os.path.exists`. `access`
		// was avoided because on Windows it reports the reparse point of a
		// dangling symlink as existing, breaking cross-language parity.
		await stat(filePath);
		return true;
	} catch {
		return false;
	}
}
